src/mds/CInode.cc

```diff
@@ -4027,21 +4027,38 @@ void CInode::decode_import(bufferlist::const_iterator& p,
 
 void InodeStoreBase::dump(Formatter *f) const
 {
+  f->open_object_section("inode");
```

Comment:
Don't think this separate section is a good thing. It break existing tools and logically it should be part of this InodeStoreBase section.

Reply:
@batrick oh. 
I cannot  find the tools that rely to this logic.  I decode the dump file using JSONDecoder.
like this:
```
JSONDecoder::decode_json("inode", inode, obj, true);
=> 
template<class T>
bool JSONDecoder::decode_json(const char *name, T& val, JSONObj *obj, bool mandatory)
```
that need all info of inode_t are in a separate  section when dump it I think.

Reply:
Well, our testing does rely on it:

````
2019-07-20T09:29:57.426 INFO:tasks.cephfs_test_runner:======================================================================
2019-07-20T09:29:57.426 INFO:tasks.cephfs_test_runner:ERROR: test_deep_split (tasks.cephfs.test_fragment.TestFragmentation)
2019-07-20T09:29:57.426 INFO:tasks.cephfs_test_runner:----------------------------------------------------------------------
2019-07-20T09:29:57.426 INFO:tasks.cephfs_test_runner:Traceback (most recent call last):
2019-07-20T09:29:57.426 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-pdonnell-testing-20190719.231534/qa/tasks/cephfs/test_fragment.py", line 193, in test_deep_split
2019-07-20T09:29:57.426 INFO:tasks.cephfs_test_runner:    frags = self.get_dir_ino("/splitdir")['dirfrags']
2019-07-20T09:29:57.426 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-pdonnell-testing-20190719.231534/qa/tasks/cephfs/test_fragment.py", line 25, in get_dir_ino
2019-07-20T09:29:57.427 INFO:tasks.cephfs_test_runner:    if ino['ino'] == dir_inono:
2019-07-20T09:29:57.427 INFO:tasks.cephfs_test_runner:KeyError: 'ino'
````

/ceph/teuthology-archive/pdonnell-2019-07-20_01:58:35-fs-wip-pdonnell-testing-20190719.231534-distro-basic-smithi/4131260/teuthology.log

These extra object sections should be removed.